routes/api/businesses.js

```diff
@@ -47,22 +46,31 @@ router.post("/", async (req, res) => {
 
 // Update
 router.put("/:id", async function (req, res) {
-  let { visible, displayName, description } = req.body
+  let {  displayName, description } = req.body
   await Business.findOne({ _id: req.params.id }, async (err, business) => {
     if (err) throw err
+    
+    if (displayName) {
+      await TextString.findOneAndUpdate({ _id: business.displayName }, displayName, {new: true, useFindAndModify: false})
+        .then(() => delete req.body.displayName)
+        .catch((err) => {throw err});
+      
+  }
 
-    if (displayName) await TextString.findOneAndUpdate({ _id: business.displayName }, displayName, {new: true, useFindAndModify: false})
-    .catch((err) => {throw err});
-
-    if (description) await TextString.findOneAndUpdate({ _id: business.description }, description, {new: true, useFindAndModify: false})
-    .catch((err) => {throw err});
-
-  })
-  if (visible) {
-    Business.findOneAndUpdate({ _id: req.params.id }, { visible }, { new: true, useFindAndModify: false })
+    if (description) {
+      await TextString.findOneAndUpdate({ _id: business.description }, description, {new: true, useFindAndModify: false})
```

Comment:
Following a previous comment of mine, once you will split the code to `routes` and `api`, then you can create a `api/textString.js` file, which will have an `update` func.
Since this is a repeating logic, this function will just get an `id`, and do the whole
`TextString.findOneAndUpdate...` thing, including removing the previous one